rpcs3/Emu/IdManager.h

```diff
@@ -559,10 +559,8 @@ class idm
 				place = object;
 				return object;
 			}
-			else
-			{
-				key_ptr->clear();
-			}
+
+			key_ptr->clear();
```

Comment:
If you are already here, it should be:
```suggestion
			*key_ptr = {};
```